src/settings.rs

```diff
@@ -204,6 +208,23 @@ impl Settings {
     }
 }
 
+// TODO: someone better at lifetimes could fix these.
+// The returned settings need only last as long as the request.
+// we clone to break out of the Arc
+impl From<&HttpRequest> for Settings {
```

Comment:
This was trickier than I thought it was gonna be 😅 but I think this works:

https://github.com/mozilla-services/contile/compare/feat/145-trace...feat/145-trace-lifetimes

Wanna apply the same to `impl From<&ServiceRequest> for Settings` below?

Reply:
Oh, cool! Thanks!
Yep, I'll modify the other ones along the same lines.